### ISSUE TITLE
Misleading docs or wrong implementation for LeftCurly ignoreEnums

### ISSUE THREAD

**Issue by Vampire** [checkstyle/checkstyle#6459]
The description suggests that with `option = eol` and `ignoreEnums = true`, enums are ignored.
I actually wondered about this, because one could simply leave out the `ENUM_DEF` token to achieve that.
But as I then looked deeper I found that it only cares about one aspect, that
```java
enum Foo { BAR, BAZ }
```
is allowed when the rules would mandate
```java
enum Foo {
    BAR,
    BAZ
}
```
while this will still trigger an error
```java
enum Foo
{
    BAR,
    BAZ
}
```

The only thing that is allowed is, that after the Lcurly is not immediately following a newline character for enum def, to allow such one-line enum defs I'd guess.
The docs should make this clearer. :-)
